client.go

```diff
@@ -108,20 +108,27 @@ func (c *Client) invokePlugin(ctx context.Context, name string, r *types.Request
 	cmd.Stderr = os.Stderr
 
 	out, err := cmd.Output()
+	msg := "output:"
+	if len(out) > 0 {
+		msg = fmt.Sprintf("%s \"%s\"", msg, out)
+	} else {
+		msg = fmt.Sprintf("%s <empty>", msg)
+	}
 	if err != nil {
-		// ExitError is returned when there is a non-zero exit status
-		if _, ok := err.(*exec.ExitError); ok {
-			var pe types.PluginError
-			if err := json.Unmarshal(out, &pe); err != nil {
-				return nil, errors.Wrapf(err, "%s: %s", name, out)
-			}
-			return nil, &pe
+		if exitErr, ok := err.(*exec.ExitError); ok {
+			// ExitError is returned when there is a non-zero exit status
+			msg = fmt.Sprintf("%s exit code: %d", msg, exitErr.ExitCode())
+		} else {
+			// plugin did not get a chance to run, return exec err
+			return nil, err
 		}
-		return nil, errors.Wrapf(err, "%s: %s", name, out)
 	}
 	var result types.Result
 	if err := json.Unmarshal(out, &result); err != nil {
-		return nil, err
+		return nil, errors.Errorf("failed to unmarshal plugin output: %s: %s", err.Error(), msg)
```

Comment:
how do you fill about: 

`errors.Errorf("...  output: %q , exit code: %d", err.Error(), out, cmd.ProcessState.ExitCode())`

in worst case it'll be `output: "", exit code: -1`, but will remove if branches above at all for string formatting.

Reply:
`output: ""` was something I wanted to avoid in the error messages as that makes the debug logs in containerd a little unreadable with escapes - `output: \"\"` and may make the person reading the log wonder if something was not captured. The `output: <empty>`  is relatively more clear and authoritative.